storage/tests/unit/test_blob.py

```diff
@@ -576,7 +583,7 @@ def test_delete(self):
         bucket._blobs[BLOB_NAME] = 1
         blob.delete()
         self.assertFalse(blob.exists())
-        self.assertEqual(bucket._deleted, [(BLOB_NAME, None)])
+        self.assertEqual(bucket._deleted, [(BLOB_NAME, None, blob.generation)])
```

Comment:
Please duplicate this test, once with an explicit generation, and make this assertion be clear for both cases (`None` for the case where the blob is created without a generation).